### PR TITLE
jsonrpsee: add host and origin filtering

### PR DESCRIPTION
**Pull request by niklasad1**

### CODE REVIEW

File: client/rpc-servers/src/lib.rs [hunk 1]
```diff
@@ -99,8 +99,24 @@ pub fn start_http<M: Send + Sync + 'static>(
 		.map(|mb| mb.saturating_mul(MEGABYTE))
 		.unwrap_or(RPC_MAX_PAYLOAD_DEFAULT);
 
+	let mut acl = AccessControlBuilder::new();
+
+	log::info!("starting JSONRPC HTTP server: addr={}, cors={:?}", addr, cors);
+
+	if let Some(cors) = cors {
+		// Whitelist listening address.
+		let host = Host::parse(&format!("localhost:{}", addr.port()));
```
Comment (niklasad1) [inline client/rpc-servers/src/lib.rs:108]: Incredible ugly and annoying, we should fix this API.

File: frame/merkle-mountain-range/rpc/Cargo.toml [hunk 2]
```diff
@@ -17,6 +17,7 @@ codec = { package = "parity-scale-codec", version = "2.0.0" }
 jsonrpsee = { git = "https://github.com/paritytech/jsonrpsee", branch = "master", features = ["server"] }
 serde_json = "1"
 serde = { version = "1.0.126", features = ["derive"] }
+log = "0.4"
```
Comment (dvdplm) [inline frame/merkle-mountain-range/rpc/Cargo.toml:20]: ah so this is what you meant by the issue on `log`? Yeah this is a bit awkward. :/

**Review by dvdplm — APPROVED** [review]
I'm ok merging this for now and loop back and change it up once we sort out the `set_allowed_hosts` for http and the `log` requirement for using proc macros.

File: frame/merkle-mountain-range/rpc/Cargo.toml (hunk 2 above)
Comment (niklasad1) [reply frame/merkle-mountain-range/rpc/Cargo.toml:20]: yeah :)

File: client/rpc-servers/src/lib.rs [hunk 3]
```diff
@@ -99,8 +99,24 @@ pub fn start_http<M: Send + Sync + 'static>(
 		.map(|mb| mb.saturating_mul(MEGABYTE))
 		.unwrap_or(RPC_MAX_PAYLOAD_DEFAULT);
 
+	let mut acl = AccessControlBuilder::new();
+
+	log::info!("starting JSONRPC HTTP server: addr={}, cors={:?}", addr, cors);
+
+	if let Some(cors) = cors {
```
[the remark this answers is not in the document]
Comment (niklasad1) [reply client/rpc-servers/src/lib.rs:106]: All hosts/origins are enabled by default: https://github.com/paritytech/jsonrpsee/blob/master/http-server/src/access_control/mod.rs#L115

We should probably document it clearly in jsonrpsee I guess

**Pull request merged**